src/javascript/app/App/Components/Elements/ContractDrawer/contract-drawer.jsx

```diff
@@ -171,7 +172,7 @@ class ContractDrawer extends Component {
             <div className={classNames('contract-drawer', {})}>
                 <div
                     className='contract-drawer__heading'
-                    onClick={() => this.props.history.goBack()}
+                    onClick={() => this.props.history.push(routes.reports)}
```

Comment:
Will it go back properly based on where we came from in Reports? e.g - `Profit Table`, `Statement`, `Open Positions`

Reply:
True, it won't. The issue is if the user pastes the contract route in the url, then presses go back it will redirect to trade. Let me check if there is a better way to handle this

Reply:
Thanks, because I think in most cases users will be going to `contract` route from one of the three sub-routes in Reports. One way to check if they're from reports or not is whether the `contract_id` prop exists or not. It only fallbacks to contract in `URL` if the `contract_id` `prop` is undefined.